packages/short-bind/src/core/transformer.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  type ConstantTypes,
+  type NodeTransform,
+  type NodeTypes,
+  createSimpleExpression,
+  processExpression,
+} from '@vue/compiler-core'
+
+const reg = /^(::?|\$|\*)(?=[A-Z_a-z])/
+
+export interface Options {
+  // empty
+}
+
+// eslint-disable-next-line unused-imports/no-unused-vars -- not be used by now
+export function transformShortBind(_options: Options = {}): NodeTransform {
+  return (node, context) => {
+    if (node.type !== (1 satisfies NodeTypes.ELEMENT)) return
+
+    for (const prop of node.props) {
+      if (
+        reg.test(prop.loc.source) &&
+        (prop.type === (6 satisfies NodeTypes.ATTRIBUTE)
+          ? !prop.value
+          : prop.type === (7 satisfies NodeTypes.DIRECTIVE)
+          ? !prop.exp
+          : false)
+      ) {
+        const str = `${prop.loc.source.replace(reg, '')}`
+        let [str1, str2] = str.split('-')
+        str2 = str2 ? str2.charAt(0).toUpperCase() + str2.slice(1) : ''
+        const valueName = str1 + str2
+
+        if (prop.type === (6 satisfies NodeTypes.ATTRIBUTE)) {
+          prop.value = {
+            type: 2 satisfies NodeTypes.TEXT,
+            content: valueName,
+            loc: {
+              start: {
+                offset: prop.loc.start.offset,
+                column: prop.loc.start.column,
+                line: prop.loc.start.line,
+              },
+              end: {
+                offset: prop.loc.end.offset,
+                column: prop.loc.end.column,
+                line: prop.loc.end.line,
+              },
```

Comment:
If position is not changed, pass the raw value directly. `prop.loc.start` and `prop.loc.end`